website/source/docs/guides/semaphore.html.md

```diff
@@ -65,75 +67,106 @@ This is something like:
+    "ModifyIndex": 897
+  }
+]
+```
+Note that the `Value` we embedded into `<lock>` is Base64 encoded when returned by the API.
+
+When the `<lock>` is read and its `Value` is decoded, we can verify the remote `Limit` agrees with the `Holders` count. 
+This is used to detect a potential conflict. The next step is to determine which of the current
+slot holders are still alive. As part of the results of the `GET`, we also have all the contender
 entries. By scanning those entries, we create a set of all the `Session` values. Any of the
 `Holders` that are not in that set are pruned. In effect, we are creating a set of live contenders
 based on the list results and doing a set difference with the `Holders` to detect and prune
-any potentially failed holders.
+any potentially failed holders. In this example `<session>` is present in `Holders` and 
+is attached to the key `<prefix>/<session>`, so no pruning is required.
 
-If the number of holders (after pruning) is less than the limit, a contender attempts acquisition
-by adding its own session to the `Holders` and doing a Check-And-Set update of the `<lock>`. This
-performs an optimistic update.
+If the number of holders after pruning is less than the limit, a contender attempts acquisition
+by adding its own session to the `Holders` list and doing a Check-And-Set update of the `<lock>`. 
+This performs an optimistic update.
 
-This is done by:
+This is done with:
 
 ```text
-curl -X PUT -d <Updated Lock> http://localhost:8500/v1/kv/<lock>?cas=<lock-modify-index>
+curl -X PUT -d <Updated Lock Body> http://localhost:8500/v1/kv/<lock>?cas=<lock-modify-index>
  ```
+`lock-modify-index` is the latest `ModifyIndex` value known for `<lock>`, 901 in this example.
 
-If this succeeds with `true`, the contender now holds a slot in the semaphore. If this fails
-with `false`, then likely there was a race with another contender to acquire the slot.
-Both code paths now go into an idle waiting state. In this state, we watch for changes
-on `<prefix>`. This is because a slot may be released, a node may fail, etc.
-Slot holders must also watch for changes since the slot may be released by an operator
-or automatically released due to a false positive in the failure detector.
+If this request succeeds with `true`, the contender now holds a slot in the semaphore. 
+If this fails with `false`, then likely there was a race with another contender to acquire the slot.
 
-Note that the session by default makes use of only the gossip failure detector. That
-is, the session is considered held by a node as long as the default Serf health check
-has not declared the node unhealthy. Additional checks can be specified if desired.
+To re-attempt the acquisition, we watch for changes on `<prefix>`. This is because a slot
+may be released, a node may fail, etc. Slot holders must also watch for changes since the
+slot may be released by an operator or automatically released due to a false positive
+in the failure detector.
```

Comment:
I'd reword this to point out that a holder MUST watch the lock value constantly (i.e. it must be watching _before_ it starts it's critical work/assumes lock is held). You say that but it's second so I almost skipped over it.

There are several things the watch achieves:

 1. If the operator or something changes the lock state then the holder should consider their lock lost
 2. Similar to above but if the failure detector fails for some fuzzy reason (packet loss, CPU overload etc) and the Consul server considers the session closed even though the process is still running, the process needs to know about it.

So the watch is important to guarantee that the process still holds the lock - it should consider it's lock revoked if either the watch fails to connect for a period (while it is retried, lock is lost) or if it gets an update that doesn't have the node in the holders list any more.